KAS-3197 - bugfix/KAS-3197-treatments-ignore-cache

ValenberghsSven opened the pull request:
Bug:

Yggdrasil is not sending a cache clear signal to `agendaitem/id/treatments` on "releasing decisions"
In some situations this can lead to cache never invalidating unless extra steps are done and data not showing
(f.e. when government profile checked that agenda before the decisions were released they effectively cached the result as empty, so without a cache invalidation this stays empty)

"fixed" it by ignoring cache for this specific call.

I'm guessing we need some documentation here, if the fix is accepted.
Also, when we separate the treatments and decisions, we might have to keep this in mind.
Perhaps we should do something similar to documents, where we release the pieces but not the files, not sure if this is acceptable for the business. But we could pair that with a conditional flag in frontend to check for "releasedDecisions == true".

erikap commented:
Can you amend the first paragraph of your PR description to the commit message?  Might help in the future when we're looking for the reason to bypass cache.

Ok for me to accept as temporary workaround, but I would like to provide time (in another story probably @tdn) to setup a reproducable case such that we can report an issue on mu-cache/mu-cl-resources to get it fixed. 

Do you have an estimation about the performance impact? Because the change affects all users, not only the ones for which the bug was fixed.

ValenberghsSven committed:
KAS-3197 add dispatcher rules to bypass cache for 1 path

Yggdrasil is not sending a cache clear signal to agendaitem/id/treatments on "releasing decisions"
In some situations this can lead to cache never invalidating unless extra steps are done and data not showing
(f.e. when government profile checked that agenda before the decisions were released they effectively
cached the result as empty, so without a cache invalidation this stays empty)

ValenberghsSven commented:
messed up a bit adding the commit messages (ended up with double commits and multiple merges) so just forced it back to a single commit (in case anyone checked it out locally, which I doubt)

Don't really have numbers on performance.
Skipping cache for the relation was 1 possible temporary fix.
Might not be the way to go.
I could run jenkins a couple of time but is not really an indication of an agenda with 100 items.

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/app-kaleidos into bugfix/KAS-3197-treatments-ignore-cache

ValenberghsSven commented:
Should we still do this with the remodeling in mind?

erikap commented:
> Should we still do this with the remodeling in mind?

I would park it for now and re-evaluate after the treatment/decision-activity data migration tbh.

ValenberghsSven commented:
Closing this PR and removing the branch.
This temp fix will most likely no longer work or will no longer be needed after the upcoming refactor
Also, the use case of this happening is low and can be "fixed" per occurrence by saving one of the agendaitems in the agenda.
